### ISSUE TITLE
warning when sampling grid is large

### ISSUE THREAD

**Issue by johnnygreco** [ArtificialStellarPopulations/ArtPop#2]
We should add a user warning when the sampling grid size of `xy_from_grid` reaches some large value.

**Comment by irvin2492** [comment]
I would like to work on this. Is there a certain dimension size value at which the sample grid should raise a warning or should the warning be used to keep the user from inputting large values?

**Comment by johnnygreco** [comment]
Thanks for your interest @irvin2492! 

This is somewhat arbitrary, but I'd say it would be good to warn the user if `image`, defined on [line 54](https://github.com/ArtificialStellarPopulations/ArtPop/blob/main/src/artpop/space/sample.py#L54), will take up more than 1 Gb of memory. Assuming `np.float64` pixels, this will happen when `memory_in_Gb > 1`, where `memory_in_Gb` is defined like this:
```
npix = 2 * sample_dim**2
byte_per_pixel = 8 # for np.float64
memory_in_Gb = byte_per_pixel * npix / 1e9
```
And the check should happen _before_ ``xx`` and ``yy`` are created.

**Comment by johnnygreco** [comment]
Issue closed by #3.
